textdb/textdb-perftest/src/main/java/edu/uci/ics/textdb/perftest/excelsink/ExcelSink.java

```diff
@@ -0,0 +1,164 @@
+
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+import java.io.File;
+import edu.uci.ics.textdb.api.constants.SchemaConstants;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.api.dataflow.ISink;
+import edu.uci.ics.textdb.api.exception.TextDBException;
+import edu.uci.ics.textdb.api.field.IField;
+import edu.uci.ics.textdb.api.schema.Schema;
+import edu.uci.ics.textdb.api.tuple.Tuple;
+import edu.uci.ics.textdb.api.utils.Utils;
+import edu.uci.ics.textdb.perftest.utils.PerfTestUtils;
+
+/**
+ * ExcelSink is a sink that can write a list of tuples into an excel file
+ * setInputOperator -> open -> collectAllTuples -> close -> deleteFile
+ * The path of saved files is "textdb/textdb/textdb-perftest/src/main/resources/index/excel/"
+ * @author Jinggang Diao
+ *
+ */
+public class ExcelSink implements ISink {
+    
+    private IOperator inputOperator;
+    
+    private Schema inputSchema;
+    private Schema outputSchema;
+    
+    private boolean isOpen;
+    
+    private Workbook wb;
+    private FileOutputStream fileOut;
+    Sheet sheet;
+    Row row;
+    public static String excelIndexDirectory = PerfTestUtils.getResourcePath("/index/excel/") + "/";
+    private String fileName;
+
+    
+    public ExcelSink() {
+    	isOpen = false;
+    }
+    
+    public boolean getIsOpen(){
+    	return isOpen;
+    }
+    
+    public void setInputOperator(IOperator inputOperator) {
+        this.inputOperator = inputOperator;
+    }
+    
+    public IOperator getInputOperator() {
+        return this.inputOperator;
+    }
+
+    @Override
+    public Schema getOutputSchema() {
+        return outputSchema;
+    }
+
+    @Override
+    public void open() throws TextDBException{
+        if (isOpen) {
+            return;
+        }     
+        inputOperator.open();
+        inputSchema = inputOperator.getOutputSchema();
+        outputSchema = Utils.removeAttributeFromSchema(inputSchema, SchemaConstants._ID, SchemaConstants.PAYLOAD);
+        try{
+            wb = new XSSFWorkbook();
+            DateFormat df = new SimpleDateFormat("YYYYMMDD_HH_mm_ss_SSSS");
+            Date dateobj = new Date();
+            fileName = df.format(dateobj) + ".xlsx";
+        	fileOut = new FileOutputStream(excelIndexDirectory + fileName);
+        	sheet = wb.createSheet("new sheet");
+        	row = sheet.createRow((short)0);
+        	List<String> attributeNames = outputSchema.getAttributeNames();
+        	//TODO::if attribute is a list, we cannot give it a fixed column size
+        	for(int i = 0; i < attributeNames.size(); i++){
+        		String attributeName = attributeNames.get(i);
+            	row.createCell(i).setCellValue(attributeName);
+        	}
+            isOpen = true;
+        }catch(Exception e){
+        	System.out.println("ExcelSink::open  "+e.getMessage());
```

Comment:
DO NOT catch a general `Exception` and only print it error message.

Catch specific exceptions, for example, `IOException`, and re-throw an `DataFlowException`